### ISSUE TITLE
Feat - Allow disabling parts of intellisense

### ISSUE THREAD

**Issue by PowerUser64** [DesModder/DesModder#827]
Intellisense is a really nice feature, but I find that parts of its features get in the way some times, so I would like to disable them. 

Such features include:

- Press enter to complete
- Show completion menu for variable and function names
- Show completion menu for function signatures

Having options for these would prevent intellisense from getting in the way in certain situations.

---

### Examples where each of these options would be nice

Example 1: if you want to press enter to have desmos automatically add declarations for missing variables, but desmodder makes it autocomplete a variable instead
![image](https://github.com/DesModder/DesModder/assets/24578572/fbb24335-f653-44eb-bb12-34879e7f750c)

Example 2: I find it is very rare that I don't know the name of a variable or function I'm looking for, so I would like desmodder to only complete function signatures, as I find positional arguments and data types to be harder to keep track of.

Example 3: Because the previous example is subjective, and I suppose the opposite type of person does exist, I think it would be good to have a way to enable/disable showing completions for positional arguments.

**Comment by jared-hughes** [comment]
I agree that adding options would help here, but it could be better to make these behaviors work better by default. In particular, I suggest:
- Enter, Up, and Down should always do the vanilla behavior by default (without needing to press Esc first)
- Have to press "Tab" to enter the intellisense menu, at which point Enter, Up, and Down behave like they donow.
This handles two of the issues. I've split this off to https://github.com/DesModder/DesModder/issues/828.

So this leaves the ability to toggle "Show completion menu for variable and function names" and "Show completion menu for function signatures." You observe usefully that there's really two _separate_ behaviors here:
- Completions when the cursor is before the `(` -- this is what gets in the way
- Signature notes when the cursor is after the `)` -- I suspect this doesn't get in the way for anyone, and I'd rather not add more options just for the sake of more options.
